spring-cloud-stream-metrics/src/main/java/org/springframework/cloud/stream/metrics/config/MetricJsonSerializer.java

```diff
@@ -40,7 +40,12 @@
 @JsonComponent
 public class MetricJsonSerializer {
 
-	final static DateFormat df = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'");
+	final static ThreadLocal<DateFormat> df = new ThreadLocal<DateFormat>(){
+			@Override
+			protected DateFormat initialValue() {
+					return new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'");
+			}
+	};
```

Comment:
My comment was specifically related to `we may need to change the formatter itself, on the collector we changed it to round values to 2 decimal places`. While I admit I don't know exactly what was done in the collector, what I intended to say was that that if we consider changing the format, it should be done  only as a configuration option and preserving the current behaviour as default. Otherwise I'm quite happy not to configure anything.